### PR TITLE
fix appointment bugs

### PR DESCRIPTION
**Pull request by simbayippy**
This PR contains fixes for 7 main things

### A:
Appointment message for add, delete, edit Appointments now shows the appropriate message upon success 

### B:

edit_appointment command now fixed to check for appropriate cases of:
1. if only description changed, do not check if doctor & patient has same appointment
2. if time of appointment changed, check if doctor & patient has appointments on those times
3. if both changed, check if this Appointment already exists

### C:
updated doctor's appointment list to contain edited appointment after calling edit_appt command. used to cause major bug that breaks application when the following commands are done in sequence
1. edit appointment details
2. attempt to delete appointment. break caused, as doctor's appointment list did not contain the updated appointment

### D:
fixed issue when editing a person's name does not change the appointments name

### E:
fixed issue where after editing a person, his appointment details are lost in db.

### F:
updated add appointment command to only accept timings between 9am to 5pm.

### G:
changed appointments such that each timeslot can only have 1 appointment. i.e. 2 different doctors cannot have appointments on the same timeslot

## Issues fixed and closed:
#125 #126 #128 #131 #133 #136 #139 #147 #149 #151 #153 #154 #156 #159 #163 #164

### CODE REVIEW

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -93,19 +93,16 @@ public static String formatDoctor(Doctor doctor) {
...
-                .append("\n")
-                .append("Description: ")
+        builder.append(" Patient: ")
+                .append(appointment.getPatientName())
+                .append(" | Description: ")
                 .append(appointment.getDescription())
-                .append("\n")
-                .append("Date: ")
+                .append(" | Date: ")
                 .append(appointment.getDateTime());
```
Comment (tohpinren) [inline src/main/java/seedu/address/logic/Messages.java]: Can you change this appointment datetime string to "01 Jan 2024 09.00AM" format?

File: src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java [hunk 2]
```diff
@@ -80,6 +82,9 @@ public CommandResult execute(Model model) throws CommandException {
         String name = targetDoctor.getName().toString();
         Appointment toAdd = new Appointment(description, dateTime, targetPatient, name);
 
+        if (targetPatient.hasAppointmentOnTimeslot(toAdd)) {
+            throw new CommandException(MESSAGE_SAME_APPOINTMENT_TIME);
+        }
         if (targetPatient.hasAppointment(toAdd)) {
```
Comment (tohpinren) [inline src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java:88]: image

If previous add appointment command was an error, successful command -> error message still stays for some reason

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java [hunk 3]
```diff
@@ -82,13 +84,31 @@ public CommandResult execute(Model model) throws CommandException {
         Appointment editedAppointment = createEditedAppointment(appointmentToEdit, editAppointmentDescriptor,
```
Comment (tohpinren) [inline src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java:83]: image


Getting seedu.address.model.timeslots.exceptions.TimeSlotNotFoundException when editing appointment time

**Review by tohpinren — CHANGES_REQUESTED** [review]
Still some big issues with the bugs - especially regarding edit appointment

File: src/main/java/seedu/address/logic/Messages.java (hunk 1 above)
Comment (simbayippy) [reply src/main/java/seedu/address/logic/Messages.java]: sure! just added

File: src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java (hunk 2 above)
Comment (simbayippy) [reply src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java:88]: need the steps to recreate the problem to get a better insight into whats causing the issue

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java (hunk 3 above)
Comment (simbayippy) [reply src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java:83]: similarly will require the steps to recreate the issue

File: src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java (hunk 2 above)
Comment (simbayippy) [reply src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java:88]: issue seems to be created because of timeslotnotfoundexception

Comment (simbayippy) [reply src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java:88]: Screenshot 2023-11-07 at 6 14 22 PM

so, similar to pin ren, i can successfully add a new appointment, but error is thrown in the highlighted 2 lines. @wj331 can you provide insights into what this does? and why is it throwing a TimeSlotNotFoundException 

in addAppointmentCommand

update: nvm. see telegram

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java (hunk 3 above)
Comment (simbayippy) [reply src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java:83]: solved in latest commit

**Review by tohpinren — APPROVED** [review]

**Comment by tohpinren** [comment]
LGTM previous issues were fixed

**Pull request merged**